frontend/src/components/EventDeleteButton.jsx

```diff
@@ -0,0 +1,57 @@
+import "../styles/EventDeleteButton.css";
+import { Trash } from "lucide-react";
+import { useNavigate } from "react-router";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import axios from "axios";
+import { ToastContainer, toast, Slide } from "react-toastify";
+
+const EventDeleteButton = ({ eventId }) => {
+    const navigate = useNavigate();
+
+    const queryClient = useQueryClient();
+
+    const deleteEventMutation = useMutation({
+        mutationFn: (eventId) => {
+            const url = import.meta.env.VITE_SERVER_API;
+            return axios.delete(`${url}/events/${eventId}`, {
+                withCredentials: true,
+            });
+        },
+        onSuccess: () => {
+            console.log("Success");
+            queryClient.invalidateQueries("events");
+            navigate("/feed");
+        },
+        onError: () => {
+            toast.error("Unknown Error... Try again later");
```

Comment:
If you're able to get this to show (ex. temporarily change the backend to always throw), it'd be good to include in the test plan!